scintillometry/utils/phase_utils.py

```diff
@@ -0,0 +1,135 @@
+# Licensed under the GPLv3 - see LICENSE
+
+import warnings
+
+import numpy as np
+import astropy.units as u
+import astropy.time as time
+
+try:
+    from .pint_utils import PintToas, make_toa_list
+    from pint.models import get_model
+except ImportError:
+    pass
+from .predictor import Polyco
+
+
+__all__ = ['PintPhase', 'PolycoPhase']
+
+
+class PintPhase(object):
+    """A utility class designated for computing phase using PINT.
+       Parameter
+       ---------
+       par_file : str
+           TEMPO/TEMPO2 style parameter fileself.
+       obs : str
+           Observatory name or observatory codeself.
+       obs_freq : float or `~astropy.units.Quantity`.
+           Observing frequency default units is 'MHz'
+       ephem : str
+           The solar system ephemeris version.
+       bipm_version : str
+           The BIPM clock correction version.
+
+       Note
+       ----
+       This method provides high precision phase calculation. The fractional
+       phase is between -0.5 to 0.5.
+    """
+    def __init__(self, par_file, obs, obs_freq, ephem='de436', include_bipm=True,
+                 bipm_version='BIPM2015', include_gps=True, planets=False,
```

Comment:
I think all arguments after `bipm_version` should be removed - they can be given as `kwargs` anyway.

Reply:
I think the user should see it. Because the older version BIPM files do not cover the recent date.  I feel it is kind of dangerous to hide it. Maybe it is the time to push Astropy time object to handle it?

Reply:
OK, so let's just remove the argument beyond it, i.e., the ones not currently documented.

If it isn't there already, could you raise an issue over at astropy that describes what the BIPM things do?

Reply:
This comment still stands